### PR TITLE
CU-et6c7a Deprecates racePair and raceTriple

### PR DESCRIPTION
**Pull request by nomisRev**
This PR deprecates all code in `RacePair.kt` and `RaceTriple.kt` since they break guarantees provided with KotlinX Coroutines structured coroutines and KotlinX offers other/safer mechanisms to cover this use-case.

### CODE REVIEW

**Comment by franciscodr** [comment]
Task linked: [CU-et6c7a Deprecate extensions in racePair.kt and raceTriple.kt](https://app.clickup.com/t/et6c7a)

**Review by rachelcarmena — APPROVED** [review]
:fire:

File: arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/RacePair.kt [hunk 1]
```diff
@@ -12,11 +12,14 @@ import kotlin.coroutines.intrinsics.suspendCoroutineUninterceptedOrReturn
 import kotlin.coroutines.intrinsics.intercepted
 import kotlin.coroutines.intrinsics.COROUTINE_SUSPENDED
 
+@Deprecated("Will be removed since it leaks Fiber, and breaks structured concurrency. Replace with select")
```
Comment (aballano) [inline arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/RacePair.kt:15]: what's `select` here?

**Review by aballano — APPROVED** [review]

**Review by danimontoya — APPROVED** [review]

File: arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/RacePair.kt (hunk 1 above)
Comment (nomisRev) [reply arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/RacePair.kt:15]: https://kotlinlang.org/docs/select-expression.html

Comment (aballano) [reply arrow-libs/fx/arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/RacePair.kt:15]: Ugh, I definitely prefer the `race` term :D Thank you!

**Pull request merged**